### PR TITLE
Hl comment implementation  

### PR DESCRIPTION
**Pull request by Hann61**

### CODE REVIEW

File: src/controllers/commentController.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
+import commentService from '../services/commentService.js';
+import mongoose from 'mongoose';
+
+export const createComment = async (req, res) => {
+  try {
+    console.log('Server:: create the Comment');
+    const id = req.params.threadId;
+    if (!mongoose.Types.ObjectId.isValid(id)) {
```
Comment (ddm1293) [inline src/controllers/commentController.js]: I don't think this is necessary tho, you can try on the postman to have an invalid objectId.

File: src/controllers/commentController.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      error: err.message
+    });
+  }
+};
+
+export const getCommentsByThread = async (req, res) => {
+  try {
+    console.log('Server::Getting comments - running getComments');
+    const threadId = req.params.threadId;
+    if (!mongoose.Types.ObjectId.isValid(threadId)) {
```
Comment (ddm1293) [inline src/controllers/commentController.js]: again.

File: src/models/commentModel.js [hunk 3]
```diff
@@ -1,14 +1,29 @@
...
     type: String,
+    required: true
+  },
+  author: {
+    id: {
+      type: Schema.Types.ObjectId,
+      ref: 'User',
+      required: true
+    },
+    commenterName: String
```
Comment (ddm1293) [inline src/models/commentModel.js]: still not sure if this is necessary. u shouble be able to get this by the authorId.

**Review by ddm1293 — COMMENTED** [review]
Have u tried every api on postman? pls make sure they all work.

File: src/controllers/commentController.js (hunk 1 above)
Comment (Hann61) [reply src/controllers/commentController.js]: Without this, if we give an invalid threadId, postman reply "Cannot create this comment"; If have this, we will more clear about the error type

File: src/models/commentModel.js (hunk 3 above)
Comment (Hann61) [reply src/models/commentModel.js]: Fixed in the new commit!

File: src/controllers/commentController.js (hunk 2 above)
Comment (Hann61) [reply src/controllers/commentController.js:42]: If removed, the postman has an error like "error": "Cast to ObjectId failed for value \"6483f0e32215626486c35\" (type string) at path \"_id\" for model \"Thread\". I think it is more clear in this way.
}

**Review by ddm1293 — DISMISSED** [review]

**Review by ddm1293 — APPROVED** [review]

**Review by cathylxy — APPROVED** [review]

**Pull request merged**